velox/functions/prestosql/aggregates/tests/PrestoHasherTest.cpp

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/prestosql/aggregates/PrestoHasher.h"
+#include "velox/functions/prestosql/aggregates/tests/AggregationTestBase.h"
+
+namespace facebook::velox::aggregate::test {
+using namespace facebook::velox::aggregate;
+
+class PrestoHasherTest : public AggregationTestBase {
+ protected:
+  template <typename T>
+  void assertHash(
+      const std::vector<std::optional<T>>& data,
+      const std::vector<int64_t>& expected) {
+    auto vector = makeNullableFlatVector<T>(data);
+    assertHash(expected, vector);
+  }
+
+  template <typename T>
+  void integralTest() {
+    assertHash<T>({1, 0, 100}, {9155312661752487122, 0, -367469560765523433});
+  }
+
+  void assertHash(const std::vector<int64_t>& expected, VectorPtr vector) {
+    SelectivityVector rows(vector->size());
+    PrestoHasher hasher(*vector, rows);
+
+    BufferPtr bufferPtr =
+        AlignedBuffer::allocate<int64_t>(vector->size(), pool_.get());
+    hasher.hash(rows, bufferPtr);
+
+    auto buffer = bufferPtr->as<int64_t>();
+    rows.applyToSelected(
+        [&](auto row) { EXPECT_EQ(expected[row], buffer[row]); });
+  }
+
+  VectorPtr makeMapofArraysVector(
+      std::vector<int64_t> keys,
+      std::vector<std::vector<int64_t>> values) {
+    // Map[bigint, array[bigint]]
+    auto arrayValues = vectorMaker_.arrayVector<int64_t>(values);
+
+    auto mapKeys = makeFlatVector<int64_t>(keys);
+    auto size = mapKeys->size();
+    auto offsets = AlignedBuffer::allocate<vector_size_t>(size, pool_.get());
+    auto sizes = AlignedBuffer::allocate<vector_size_t>(size, pool_.get());
+
+    auto rawOffsets = offsets->asMutable<vector_size_t>();
+    auto rawSizes = sizes->asMutable<vector_size_t>();
+
+    // Create a vector of maps with 1 key , 1 value
+    vector_size_t offset = 0;
+    for (vector_size_t i = 0; i < size; i++) {
+      rawSizes[i] = 1;
+      rawOffsets[i] = offset;
+      offset += 1;
+    }
+
+    return std::make_shared<MapVector>(
+        pool_.get(),
+        MAP(BIGINT(), ARRAY(BIGINT())),
+        nullptr,
+        size,
+        offsets,
+        sizes,
+        mapKeys,
+        arrayValues);
+  }
+
+ private:
+  BufferPtr bufferPtr_;
+};
+
+TEST_F(PrestoHasherTest, integralTypes) {
```

Comment:
It would be nice to break up this test into multiple tests, one for each type, and make sure to test nulls as well as values at the ends of the supported value range.